sw/device/silicon_creator/lib/drivers/flash_ctrl.h

```diff
@@ -444,6 +444,22 @@ typedef struct flash_ctrl_cfg {
  */
 void flash_ctrl_data_default_cfg_set(flash_ctrl_cfg_t cfg);
 
+typedef uint32_t flash_ctrl_region_index_t;
+
+/**
+ * Prevent the given region of pages from being erased or programmed.
+ *
+ * This function takes over `MP_REGION_0` and `MP_REGION_CFG_0` to prevent the
+ * given range of pages from being overwritten.
+ *
+ * @param page_offset The index of the first page in the region to protect.
+ * @param num_pages The number of pages in the region to protect.
+ */
+void flash_ctrl_data_region_protect(uint32_t page_offset, uint32_t num_pages,
+                                    flash_ctrl_region_index_t region,
```

Comment:
nit: It looks like `region` is not used yet but I assume it will be the index of the mp_region_cfg register?